vyper/codegen/expr.py

```diff
@@ -198,7 +198,11 @@ def parse_Attribute(self):
         typ = self.expr._metadata["type"]
 
         # MyEnum.foo
-        if isinstance(typ, EnumT) and typ.name == self.expr.value.id:
+        if (
+            isinstance(typ, EnumT)
+            and isinstance(self.expr.value, vy_ast.Name)
+            and typ.name == self.expr.value.id
```

Comment:
why does this check need to be changed?

Reply:
This is because `expr.value` may not be a `Name` node. In this example where the enum is a struct member, `expr.value` is a `Subscript` node instead.

It is a different issue from the one that this PR originally fixes, but was discovered while adding functionality tests.

```
enum Foobar:
    FOO
    BAR

struct Foo:
    x: uint256
    y: uint256
    z: Foobar

@external
def bar(_baz: DynArray[Foo, 3]) -> uint256:
    sum: uint256 = 0
    for i in range(3):
        e: Foobar = _baz[i].z
        f: uint256 = convert(e, uint256)
        sum += _baz[i].x * _baz[i].y + f
    return sum
```
```
def parse_Attribute(self):
       typ = self.expr._metadata.get("type")
       if typ is not None:
           typ = new_type_to_old_type(typ)
   
       # MyEnum.foo
>       if isinstance(typ, EnumType) and typ.name == self.expr.value.id:
E       AttributeError: 'Subscript' object has no attribute 'id'

vyper/codegen/expr.py:212: AttributeError
```